core/tests/protractor/fallbacks.js

```diff
@@ -0,0 +1,90 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests of the fallbacks functionality.
+ *
+ * @author Sean Lip (sean@seanlip.org)
+ */
+
+var general = require('../protractor_utils/general.js');
+var forms = require('../protractor_utils/forms.js');
+var users = require('../protractor_utils/users.js');
+var workflow = require('../protractor_utils/workflow.js');
+var editor = require('../protractor_utils/editor.js');
+var player = require('../protractor_utils/player.js');
+
+describe('Fallbacks editor', function() {
+  it('should allow a fallback to be set and apply it correctly, and ' +
+     'disallow editing of fallbacks in read-only explorations', function() {
+    users.createUser('fallback@example.com', 'fallback');
+    users.login('fallback@example.com');
+
+    workflow.createExploration('sums', 'maths');
+    general.getExplorationIdFromEditor().then(function(explorationId) {
+      editor.enableFallbacks();
+
+      editor.setStateName('card 1');
+      editor.setContent(forms.toRichText('this is card 1'));
+      editor.setInteraction('NumericInput');
+      editor.addResponse('NumericInput', null, 'card 2', true, 'Equals', 21);
+      editor.setDefaultOutcome(forms.toRichText('try again'), null, false);
+      editor.addFallback(
+        2, forms.toRichText('fallback triggered'), 'final card', true);
+
+      editor.moveToState('card 2');
+      editor.setInteraction('Continue', 'click here');
+      editor.setDefaultOutcome(null, 'final card', false);
+
+      editor.moveToState('final card');
+      editor.setInteraction('EndExploration');
+
+      editor.setObjective('To test fallbacks.');
+      editor.saveChanges();
+      workflow.publishExploration();
+
+      // Login as another user and verify that the exploration editor does not
```

Comment:
I think then that all things of this kind should be moved into privileges.js, since "checking the editor works" and "checking the editor cannot be used by unauthorised people" are fundamentally different things. However it seems reasonable to leave this here for now until the tests are refactored.